From acd46c26cca3a937dfe7098485ded1208182b1ce Mon Sep 17 00:00:00 2001
From: Rob Crittenden <rcritten@redhat.com>
Date: Tue, 6 Nov 2012 17:31:11 -0500
Subject: [PATCH] Handle the case where there are no replicas with list-ruv

This assumed that at least was returned by LDAP. This is not the case
if no replicas have ever been created.

https://fedorahosted.org/freeipa/ticket/3229
---
 install/tools/ipa-replica-manage | 21 +++++++++++----------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/install/tools/ipa-replica-manage b/install/tools/ipa-replica-manage
index 270095b8849c092ce60fa6cedccda5c0659bb5c0..d7e317c5a09d24c63b1a890c98c0a0c621050ae2 100755
--- a/install/tools/ipa-replica-manage
+++ b/install/tools/ipa-replica-manage
@@ -324,16 +324,17 @@ def get_ruv(realm, host, dirman_passwd):
         sys.exit(0)
 
     servers = []
-    for ruv in entries[0][1]['nsds50ruv']:
-        if ruv.startswith('{replicageneration'):
-            continue
-        data = re.match('\{replica (\d+) (ldap://.*:\d+)\}(\s+\w+\s+\w*){0,1}', ruv)
-        if data:
-            rid = data.group(1)
-            (scheme, netloc, path, params, query, fragment) = urlparse.urlparse(data.group(2))
-            servers.append((netloc, rid))
-        else:
-            print "unable to decode: %s" % ruv
+    for e in entries:
+        for ruv in e[1]['nsds50ruv']:
+            if ruv.startswith('{replicageneration'):
+                continue
+            data = re.match('\{replica (\d+) (ldap://.*:\d+)\}(\s+\w+\s+\w*){0,1}', ruv)
+            if data:
+                rid = data.group(1)
+                (scheme, netloc, path, params, query, fragment) = urlparse.urlparse(data.group(2))
+                servers.append((netloc, rid))
+            else:
+                print "unable to decode: %s" % ruv
 
     return servers
 
-- 
2.4.3

